docs/articles/concepts/kafka.md

```diff
@@ -0,0 +1,167 @@
+Bytewax ships with connectors for [Kafka](https://www.confluent.io/) and
+Kafka compatible systems, like [Redpanda](https://redpanda.com/).
+
+In this section, we'll discuss these connectors in detail, as well as
+provide some important operational information about using them as sources
+and sinks.
+
+## Connecting to Kafka
+
+Bytewax provides two basic ways to connect to Kafka.
+
+You can use [`KafkaSource`](/apidocs/bytewax.connectors/kafka/index#bytewax.connectors.kafka.KafkaSource)
+and [`KafkaSink`](/apidocs/bytewax.connectors/kafka/index#bytewax.connectors.kafka.KafkaSink) directly:
+
+```python
+from bytewax.connectors.kafka import KafkaSource, KafkaSink, KafkaSinkMessage
+from bytewax import operators as op
+from bytewax.dataflow import Dataflow
+
+brokers = ["localhost:19092"]
+flow = Dataflow("example")
+kinp = op.input("kafka-in", flow, KafkaSource(brokers, ["in-topic"]))
+processed = op.map("map", kinp, lambda x: KafkaSinkMessage(x.key, x.value))
+op.output("kafka-out", processed, KafkaSink(brokers, "out-topic"))
+```
+
+Or use the Kafka input [operator](/apidocs/bytewax.connectors/kafka/operators#bytewax.connectors.kafka.operators.input):
+
+```python
+from bytewax.connectors.kafka import operators as kop, KafkaSinkMessage
+from bytewax import operators as op
+from bytewax.dataflow import Dataflow
+
+brokers = ["localhost:19092"]
+flow = Dataflow("example")
+kinp = kop.input("kafka-in", flow, brokers=brokers, topics=["in-topic"])
+processed = op.map("map", kinp.oks, lambda x: KafkaSinkMessage(x.key, x.value))
+kop.output("kafka-out", processed, brokers=brokers, topic="out-topic")
+```
+
+## Error handling
+
+The [`KafkaSource`](/apidocs/bytewax.connectors/kafka/index#bytewax.connectors.kafka.KafkaSource) input
+source will raise an exception whenever an error is encountered when consuming from Kafka.
```

Comment:
This subsection lacks some sample code examples on handling exceptions in the user's code. It could be very high-level pseudo-code but a starting point is nice to have

Reply:
Well, you can't handle the exceptions inside the `KafkaSource`, but there's a `raise_on_error` option there, that is automatically handled in the operator by splitting the stream in `oks` and `errs`.
So you either let the dataflow crash on any relevant error, use `raise_on_error=False` in the source and handle the `KafkaError` type messages in a later step, or use the operator and do what you want with the error stream.

Maybe we need to clarify the options here, the `raise_on_error` parameter is never mentioned